### ISSUE TITLE
transparency has a 1 frame lag between sprites that are not siblings

### ISSUE THREAD

**Issue by mockersf** [bevyengine/bevy#1506]
## Bevy version

current main (bc4fe9b186db3f32eef38fc4241289df480fa949)

## Operating system & version

MacOS 10.15.7

## What you did

* Set the clear color to red
* Display a sprite that covers everything black as a children of another entity
* Display a sprite from a transparent png

```rust
use bevy::prelude::*;

fn main() {
    App::build()
        .add_plugins(DefaultPlugins)
        .insert_resource(ClearColor(Color::RED))
        .add_startup_system(setup.system())
        .add_system(cycle.system())
        .run();
}

fn setup(commands: &mut Commands, mut materials: ResMut<Assets<ColorMaterial>>) {
    commands
        .spawn(OrthographicCameraBundle::new_2d())
        .spawn((GlobalTransform::default(), Transform::default()))
        .with_children(|p| {
            p.spawn(SpriteBundle {
                material: materials.add(Color::BLACK.into()),
                sprite: Sprite {
                    size: Vec2::new(10000., 10000.),
                    ..Default::default()
                },
                ..Default::default()
            });
        });
}

struct ShouldTransparent;
fn cycle(
    commands: &mut Commands,
    mut cycle: Local<u32>,
    asset_server: Res<AssetServer>,
    mut materials: ResMut<Assets<ColorMaterial>>,
) {
    *cycle = (*cycle + 1) % 20;
    if *cycle == 0 {
        let texture_handle = asset_server.load("branding/icon.png");
        for x in -1..=1 {
            for y in -1..=1 {
                commands.spawn(SpriteBundle {
                    material: materials.add(texture_handle.clone().into()),
                    transform: Transform {
                        translation: Vec3::new(x as f32 * 256., y as f32 * 256., 0.1),
                        scale: Vec3::one(),
                        rotation: Quat::default(),
                    },
                    ..Default::default()
                });
                // .with(ShouldTransparent);
            }
        }
    }
}
```

## What you expected to happen

No transparency issue

## What actually happened

The red clear color leaks through for a frame
![ezgif-2-3c12a37e0f23](https://user-images.githubusercontent.com/8672791/108788534-7be40b00-7578-11eb-9708-caa142076b07.gif)

## Additional information

If I remove the hierarchy for the black background, there are no transparency issue.
If I add the tag component `ShouldTransparent` to my sprites, there are no transparency issue.

**Comment by mockersf** [comment]
as it was working on 0.4, I checked commits, and it started failing with d5a733043108972c51a785d61ad1a04c70d51d42

**Comment by alice-i-cecile** [comment]
This is probably caused by #1466 and should be checked once it's tentatively solved.

**Comment by mockersf** [comment]
fixed by #1606
